### PR TITLE
# 184948252 login using email and password 

### PR DESCRIPTION
**Pull request by Mystica52**
**What does this PR do?**
This PR display the login page
**Description of Task to be completed?**
Have the following routers working
 `/login`
    user should be able to login with the valid email and password 
**How should this be manually tested?**
     - Checkout to ft-login-email-and-password-184948250 branch
     - Install all the dependencies by running npm install
     - Then run the project by typing npm start 
     - Navigate to /login route
**Routes to be tested:**
POST /login
**What are the relevant pivotal tracker/Trello stories?**
Story ID: [184948252](https://www.pivotaltracker.com/story/show/184948252)
**Screenshoot**
image
image
image
image
image

### CODE REVIEW

**Comment by Emile-SHUMBUSHO** [comment]
From screenShots of your design I can see that there is some inconsistency in your design, try to align input field, error messages, and buttons in the same margin. 

Secondly, from your UI implementation add signIn with google option as it has been shown in your Figma design.

File: src/redux/actions/action.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export default (type,payload)=>({
+type:type,
+payload:payload
+});
```
Comment (Emile-SHUMBUSHO) [inline src/redux/actions/action.js:4]: Avoid empty line of code from your design and also align your cods in appropriate format.

File: src/redux/actions/auth.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          dispatch(action(LOGIN_SUCCESS,result));
+          console.log(result);
+
+          
+          }catch(error){
+            dispatch(action(LOGIN_FAIL,error.message));
+      console.log(error);
+          }
+    };
+};
```
Comment (Emile-SHUMBUSHO) [inline src/redux/actions/auth.js:31]: Avoid empty Line from your codes

File: src/views/login/LoginForm.jsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+          "Must Contain 8 Characters, One Uppercase, One Lowercase, One Number and One Special Case Character"
+        ),
+    }),
+      //submit form
+  onSubmit:async (values) =>{
+    dispatch(login(values.email, values.password));
+   
+
+    // console.log(values);
+    // router.push({pathname: "/success",query: values});
```
Comment (Emile-SHUMBUSHO) [inline src/views/login/LoginForm.jsx]: Avoid pushing commented codes on  a GitHub

File: src/views/login/LoginForm.jsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+  const dispatch = useDispatch();
+
+  const {loginSuccess}= useSelector((state)=>state.authState);
+  console.log(loginSuccess);
+
+  const navigation = useNavigate();
+  useEffect(()=>{
+    if(loginSuccess?.message === "User Logged Successfully"){
+      // PrivateRoutes("/dashboard");
```
Comment (Emile-SHUMBUSHO) [inline src/views/login/LoginForm.jsx]: Commented code are not allowed to be pushed on a GitHub

File: src/views/login/LoginForm.jsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      navigation("/dashboard");
+    }
+
+    
+
```
Comment (Emile-SHUMBUSHO) [inline src/views/login/LoginForm.jsx]: Empty space in your code violates a code format criteria

File: src/redux/store.js [hunk 6]
```diff
@@ -1,6 +1,17 @@
...
+    authState: authReducer
+});
+
+const store = createStore(
+    rootReducer,
+    composeEnhancers(applyMiddleware(thunk))
+);
 
 export default store;
+
```
Comment (Emile-SHUMBUSHO) [inline src/redux/store.js]: This codes performs the same functionality with codes from `reducer/store.js` please consider removing this file `rootReducer.js` and remain with `reducer/store.js`

File: src/tests/login.test.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+// import React from "react";
```
Comment (Emile-SHUMBUSHO) [inline src/tests/login.test.js]: Avoid commented code

**Review by Emile-SHUMBUSHO — CHANGES_REQUESTED** [review]

**Review by ericndungutse — APPROVED** [review]

File: src/redux/actions/auth.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        });
+        const result = await response.json();
+            localStorage.setItem("token",result.token);
+          dispatch(action(LOGIN_SUCCESS,result));
+          console.log(result);
+
+          }catch(error){
+            dispatch(action(LOGIN_FAIL,error.message));
+      console.log(error);
+          }
```
Comment (Habinezajanvier) [inline src/redux/actions/auth.js]: No need to push `console.log` in production code

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

**Review by wilbrord2 — CHANGES_REQUESTED** [review]
Screenshot_20230502-110306

Please change this error message color to white because they are not visible

**Comment by wilbrord2** [comment]
Please rebase and remove all conflicts

**Review by wilbrord2 — APPROVED** [review]

**Review by AbdulKhaliq59 — APPROVED** [review]

**Review by GasanaJr — APPROVED** [review]

**Review by Emile-SHUMBUSHO — APPROVED** [review]

**Pull request merged**